cmd/booster-http/server.go

```diff
@@ -150,33 +148,49 @@ func (s *HttpServer) handleIndex(w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(idxPage)) //nolint:errcheck
 }
 
+func (s *HttpServer) handlePieceRequest(w http.ResponseWriter, r *http.Request) {
+	q, err := url.ParseQuery(r.URL.RawQuery)
+	if err != nil {
+		msg := fmt.Sprintf("parsing query: %s", err.Error())
+		writeError(w, r, http.StatusBadRequest, msg)
+	}
+
+	// Check provided cid and format and redirect the request appropriately
+	if len(q[payloadPrefix]) == 1 && len(q["format"]) == 1 {
+		s.handleByPayloadCid(w, r)
```

Comment:
I think it will simplify the code to have all the query parsing in one place.
So I'd suggest that inside handlePieceRequest we get the query parameters, parse cids etc, and then pass those parsed variables to handleByPayloadCid and handleByPieceCid as parameters.